modules/secrets-store-vault.adoc

```diff
@@ -0,0 +1,384 @@
+* You have installed Helm.
+* You have installed HashiCorp Vault (`vault`).
+* You have configured HashiCorp Vault to store the required secrets.
+* You have access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Add the HashiCorp Helm repository by running the following command:
++
+[source,terminal]
+----
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+----
+
+. Update all repositories to ensure that Helm is aware of the latest versions by running the following command:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Create a new project for Vault by running the following command:
++
+[source,terminal]
+----
+$ oc new-project vault
+----
+
+. Install the HashiCorp Vault provider:
++
+////
+// TODO: Confirm if we still need this step to label the ns if we're doing a deployment instead of a pod
+.. Label the `vault` namespace to... TODO:
++
+[source,terminal]
+----
+$ oc label ns vault security.openshift.io/scc.podSecurityLabelSync=false pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/audit=privileged pod-security.kubernetes.io/warn=privileged --overwrite
+----
+////
+
+.. Grant privileged access to the `vault` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z vault -n vault
+----
+
+.. Grant privileged access to the `vault-csi-provider` service account by running the following command:
++
+[source,terminal]
+----
+$ oc adm policy add-scc-to-user privileged -z vault-csi-provider -n vault
+----
+
+.. Deploy HashiCorp Vault by running the following command:
++
+[source,terminal]
+----
+$ helm install vault hashicorp/vault --namespace=vault \
+  --set "server.dev.enabled=true" \
+  --set "injector.enabled=false" \
+  --set "csi.enabled=true" \
+  --set "global.openshift=true" \
+  --set "csi.daemonSet.providersDir=/var/run/secrets-store-csi-providers"
+----
+
+.. Patch the `vault-csi-driver` daemon set to set the `securityContext` to `privileged` by running the following command:
++
+[source,terminal]
+----
+$ oc patch daemonset -n vault vault-csi-provider --type='json' -p='[{"op": "add", "path": "/spec/template/spec/containers/0/securityContext", "value": {"privileged": true} }]'
+----
+
+.. Verify that the `vault-csi-provider` pods have started properly by running the following command:
++
+[source,terminal]
+----
+$ oc get pods -n vault
+----
++
+.Example output
+[source,terminal]
+----
+NAME                       READY   STATUS    RESTARTS   AGE
+vault-0                    1/1     Running   0          24m
+vault-csi-provider-87rgw   1/2     Running   0          5s
+vault-csi-provider-bd6hp   1/2     Running   0          4s
+vault-csi-provider-smlv7   1/2     Running   0          5s
+----
++
+////
+We didn't do the steps in AWS/Azure to actually add the secret data to the provider. Remove these steps in favor of the prerequisite, for consistency?
```

Comment:
@ropatil010 @dobsonj in the Azure and AWS procedures, we don't write the steps on how to actually create the secrets in the third-party secret stores. So I was planning on removing these steps. Does that make sense?

And then I added it as a prerequisite ("You have configured HashiCorp Vault to store the required secrets.") like we did for Azure/AWS. But then I wasn't sure if it actually was something that could be done prior to starting the procedure, or if it had to be done exactly at this point. 

If it has to be done right at this point, I could always put a step that is generic saying to configure the secrets in vault. Or if you guys think strongly that we show show these actual steps, it might be fine.

Let me know what you think!

Reply:
I guess you are right, we do not need to steps for Configure Hashicorp Vault if it is third-part. Lets wait for @dobsonj /Others inputs.

Reply:
I guess we need to write the steps, since we are using the secretPath: "secret/data/example1" and objectName: "testSecret1" inside secret-provider-class-vault.yaml. So i guess it may confuse if we mention the parameters inside secret-provider-xx.yaml and no were used. Lets wait for @dobsonj inputs.

Reply:
Yeah I see what you mean, but for Azure [1] for example, we reference "secret1" in the secret-provider-class-azure.yaml, but didn't have steps to say how it got there. We just had a prereq line that says you configured Azure key vault to have the required secrets configured there.

@dobsonj wdyt?

[1] https://docs.openshift.com/container-platform/4.15/nodes/pods/nodes-pods-secrets-store.html#secrets-store-azure_nodes-pods-secrets-store

Reply:
I would lean towards including a simple example like this. Adding secrets to vault needs to be done after the vault server (vault-0) is started, but before starting a pod that tries to mount the secret. So if nothing else, it shows a simple way to add secrets, and indicates when in the process it should be done.

Reply:
Okay that makes sense. It is a little bit different since it has to be done in this order, and it wouldn't really make sense to make it a prereq then. I'll go ahead and add this back in.